### PR TITLE
{CI} Remove `--exitfirst` to collect all error information at once.

### PR DESCRIPTION
**Pull request by wangzelin007**
**Related command**

**Description**

`-x, --exitfirst` means exit instantly on first error or failed test.
Remove `--exitfirst` to collect all error information at once so we can fix test issues in one go.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

**Comment by wangzelin007** [comment]
In the future we can use azdev test for consistency.
https://github.com/Azure/azure-cli-dev-tools/blob/da197478b7837f4cb89f0679c1c3115fca89fa57/azdev/operations/testtool/pytest_runner.py#L25

File: scripts/release/homebrew/test_homebrew_package.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 root_dir = '{}/lib/{}/site-packages/azure/cli/command_modules'.format(az_base, python_version)
 mod_list = [mod for mod in sorted(os.listdir(root_dir)) if os.path.isdir(os.path.join(root_dir, mod)) and mod != '__pycache__']
 
-pytest_base_cmd = 'PYTHONPATH={}/lib/{}/site-packages python -m pytest -x -v --forked -p no:warnings --log-level=WARN'.format(az_base, python_version)
```
Comment (jiasli) [inline scripts/release/homebrew/test_homebrew_package.py:20]: This line was introduced by https://github.com/Azure/azure-cli/pull/14372.

File: scripts/release/debian/test_deb_package.py [hunk 2]
```diff
@@ -10,7 +10,7 @@
 root_dir = '/opt/az/lib/python3.10/site-packages/azure/cli/command_modules'
 mod_list = [mod for mod in sorted(os.listdir(root_dir)) if os.path.isdir(os.path.join(root_dir, mod)) and mod != '__pycache__']
 
-pytest_base_cmd = '/opt/az/bin/python3 -m pytest -x -v --forked -p no:warnings --log-level=WARN'
```
Comment (jiasli) [inline scripts/release/debian/test_deb_package.py:13]: This line was introduced by https://github.com/Azure/azure-cli/pull/14843.

File: scripts/release/rpm/test_rpm_package.py [hunk 3]
```diff
@@ -11,7 +11,7 @@
 root_dir = f'/usr/lib64/az/lib/{python_version}/site-packages/azure/cli/command_modules'
 mod_list = [mod for mod in sorted(os.listdir(root_dir)) if os.path.isdir(os.path.join(root_dir, mod)) and mod != '__pycache__']
 
-pytest_base_cmd = f'PYTHONPATH=/usr/lib64/az/lib/{python_version}/site-packages python -m pytest -x -v --forked -p no:warnings --log-level=WARN'
```
Comment (jiasli) [inline scripts/release/rpm/test_rpm_package.py:14]: This line was introduced by https://github.com/Azure/azure-cli/pull/14288.

**Comment by jiasli** [comment]
I personally don't like this "`-x, --exitfirst` as the default" idea from the very beginning. `azdev` made `-x, --exitfirst` the default in https://github.com/Azure/azure-cli-dev-tools/pull/188 and introduced `--no-exit-first` argument to disable it in https://github.com/Azure/azure-cli-dev-tools/pull/191. It is opposite to the default of `pytest`.

**Review by jiasli — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Comment by wangzelin007** [comment]
> I personally don't like this "`-x, --exitfirst` as the default" idea from the very beginning. `azdev` made `-x, --exitfirst` the default in [Azure/azure-cli-dev-tools#188](https://github.com/Azure/azure-cli-dev-tools/pull/188) and introduced `--no-exit-first` argument to disable it in [Azure/azure-cli-dev-tools#191](https://github.com/Azure/azure-cli-dev-tools/pull/191). It is opposite to the default of `pytest`.

Totally agree, most of the time I will manually add `--no-exitfirst`.

**Pull request merged**